### ISSUE TITLE
Pre-evaluation period lags and returns memory allocation errors

### ISSUE THREAD

**Issue by johntikas** [tensorflow/models#7319]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS 7.6.1810, Ubuntu 18.04 (container)
- **TensorFlow installed from (source or binary)**: pip
- **TensorFlow version (use command below)**: 1.14.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: 10.1 (CUDA)
- **GPU model and memory**: GeForce GTX 1070 Ti and 8 GB
- **Exact command to reproduce**:

### Describe the problem
I am trying to train an instance segmentation model on a custom dataset using the implementation of Mask-RCNN in Tensorflow's Object Detection API. For reproducibility and deployment purposes, I am running an Ubuntu container inside a CentOS desktop, with both using the same CUDA version. Training periods seem to develop normally given the computational advantage of a CUDA-enabled GPU, nonetheless, a small time segment prior to the beginning of the evaluation takes nearly an hour before completing, while the system throws several OOM and memory allocation errors pertaining to the GPU's RAM. Perhaps, the exact source of the problem could be inferred more accurately by reviewing the log timestamps as provided below.

So far, I've experimented with different 'queue' values in both input reader configs, but without overcoming the problem and getting evaluation to complete in a matter of minutes like it does in other environments such as GCP's VM instances and Google Colab with comparable GPUs. It's important to note that the most sensitive hyperparameter to occasional out-of-memory or memory allocation errors during training periods (which are almost non-existent when using the Inception V2 architecture) is the min/max dimensions in image resizer, but these are rarely found and every single time the app will most likely terminate after a few (3-4) evaluations which typically take more than an hour or so.

Here is the log of a typical outcome of executing an estimator object's train_and_eval within Tensorflow's Object Detection API, on my own custom dataset

[tf_memory_error2.txt](https://github.com/tensorflow/models/files/3436118/tf_memory_error2.txt)

### Source code / logs
Although the dataset cannot be disclosed on a public repository, I am attaching the config file which is based on a custom dataset of eighteen (18) distinct classes, where TFRecords have been verified prior to distributing TFExamples into distinct shards. 

[mask_rcnn_inception_v2_custom.config.txt](https://github.com/tensorflow/models/files/3441968/mask_rcnn_inception_v2_custom.config.txt)

**Comment by merts1** [comment]
lower the num of examples per each batch

**Comment by gowtham-kp** [comment]
@johntikas As @merts1  mentioned, please reduce the batch size and it should work. Also please take a look at the [similar issue](https://github.com/tensorflow/models/issues/1993) and let me knpow if it helps. Thanks!

**Comment by johntikas** [comment]
@merts1 I have already set `num_examples` in the `eval_config` section to values much lower than the actual size of my eval tfrecords, like the 10 provided above, but ended up finding out that after merging a recent pull request ([#5289](https://github.com/tensorflow/models/pull/5289)) the field became deprecated, so all tfexamples are evaluated despite default or manually set values which I figured that are not regarded at all, based on my own experimentation.

Eventually, I have managed to get an evaluation session to run normally by playing around with `sample_1_of_n_eval_examples` in [model_main.py](https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py) which is also available for both train and eval `input_reader` sections as `sample_1_of_n_examples` and fundamentally represents the denominator of the fraction that takes a portion of the given examples. The final configuration that helped me attain durations of almost a couple of minutes was 10 on a total dataset of nearly 5000 eval samples, so this provides the pipeline with about 500 images.

@gowtham-kp Based on the provided config file, `batch_size` was set to 1 regarding the eval_config options, nonetheless, default options in [eval.proto](https://github.com/tensorflow/models/blob/master/research/object_detection/protos/eval.proto) have mindfully set it to 1. Typically memory errors arise as soon as I increase it over 2, something that I have also noticed for training. Recently, I stumbled upon another issue ([#4831](https://github.com/tensorflow/models/issues/4831)) which mentions that `keep_aspect_ratio_resizer` option does not allow for batching, however, I have not only found out that memory allocation errors arise also when tweaking batch size with the aspect ratio option, but also that marginally functioning values that don't return OOM like 2 don't offer significant computational advantage.

Although, I consider my case solved it would be useful to suggest any possible solution to help me reduce evaluation time without decreasing the number of examples. I've thought about converting input images to single-channel grayscale instead of RGB and messing around with resizing but would also appreciate any best practices from those who've been through the process of trial-and-error.

**Comment by sidharth1805** [comment]
https://github.com/tensorflow/models/issues/1993#issuecomment-761742221
